webrtc.html

```diff
@@ -2438,12 +2438,40 @@ <h4>
                                 <p>
                                   If a suitable transceiver was found
                                   (<var>transceiver</var> is set) and
-                                  <var>sendEncodings</var> is non-empty, set
-                                  <var>transceiver</var>.{{RTCRtpTransceiver/[[Sender]]}}.{{RTCRtpSender/[[SendEncodings]]}}
-                                  to <var>sendEncodings</var>, and set
-                                  <var>transceiver</var>.{{RTCRtpTransceiver/[[Sender]]}}.{{RTCRtpSender/[[LastReturnedParameters]]}}
-                                  to <code>null</code>.
+                                  <var>receiveEncodings</var> is non-empty, run
+                                  the following steps:
                                 </p>
+                                <ol>
+                                  <li class="needs-test">
+                                    <p>
+                                      If
+                                      <var>transceiver</var>.{{RTCRtpTransceiver/[[Sender]]}}.{{RTCRtpSender/[[SendEncodings]]}}
+                                      is non-empty, then for each encoding,
+                                      <var>encoding</var>, in
+                                      <var>transceiver</var>.{{RTCRtpTransceiver/[[Sender]]}}.{{RTCRtpSender/[[SendEncodings]]}},
+                                      copy all of <var>encoding</var>'s members
+                                      and their values to the encoding with the
+                                      same {{RTCRtpCodingParameters/rid}} in
+                                      <var>receiveEncodings</var>, if there is one.
```

Comment:
This process allows the order of RIDs applied to layers to change - old {rid: a, rid:b} can be changed to new {rid: b, rid: a}. Since these are lists, not sets, that's a change.

Reply:
Agreed. I think we need to think of this as "remote offers can remove rids from [[SendEncodings]], but cannot make other changes". An attempt to add a rid probably should result in an error, while an attempt to reorder rids might be appropriate to ignore (although throwing an error in that case might be fine too).

Reply:
The attempt to change is needed to trigger failure elsewhere.

The way things are written right now, the prose that fails is higher up in 4.5: _"If applying description leads to modifying a transceiver transceiver, and transceiver.[[[Sender]]](https://w3c.github.io/webrtc-pc/#dfn-sender).[[[SendEncodings]]](https://w3c.github.io/webrtc-pc/#dfn-sendencodings) is non-empty, and not equal to the encodings that would result from processing description, the process of applying description fails. This specification does not allow remotely initiated RID renegotiation."_

This means, oddly, that to actually trigger that failure condition requires us to (attempt to) make a change here.

Yes this is terrible action-at-a-distance spec writing that I got some good [feedback](https://github.com/w3c/webrtc-pc/pull/2758#issuecomment-1198249698) on already on how to rewrite.